Makefile missing build-cli-oci-* targets

**Bug description**

PR https://github.com/vmware-tanzu/tanzu-framework/pull/1144 added pattern creation for these targets and set them as dependencies for other targets. But no targets were added to match the generated patterns, so attempting to call one of the targets that sets these as a dependency like `make build-cli-with-oci-discovery` fails with an error:

```
make: *** No rule to make target `build-plugin-oci-linux-amd64', needed by `build-cli-with-oci-discovery'.  Stop
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

--- a/Makefile
+++ b/Makefile
@@ -206,7 +206,7 @@ endif
 
 # Dynamically generate the OS-ARCH targets to allow for parallel execution
 CLI_JOBS_OCI_DISCOVERY := $(addprefix build-cli-oci-,${ENVS})
-CLI_ADMIN_JOBS_OCI_DISCOVERY := $(addprefix build-plugin-oci-,${ENVS})
+CLI_ADMIN_JOBS_OCI_DISCOVERY := $(addprefix build-plugin-admin-oci-,${ENVS})
 
 CLI_JOBS_LOCAL_DISCOVERY := $(addprefix build-cli-local-,${ENVS})
 CLI_ADMIN_JOBS_LOCAL_DISCOVERY := $(addprefix build-plugin-admin-local-,${ENVS})

